commit 98bfb0469402f662a835ca56a26512b438ca59d3
Author: bertin <macportsuser@macports.org>
Date:   Sun May 25 22:21:49 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/patch-portprovides-support-symlinked-prefix.diff

diff --git a/src/port/port.tcl b/src/port/port.tcl
index 50789b095..f41ecdf58 100755
--- a/src/port/port.tcl
+++ b/src/port/port.tcl
@@ -2188,6 +2188,11 @@ proc action_notes { action portlist opts } {
 }
 
 
+proc macports::normalize { filename } {
+    set prefmap [list [file dirname [file normalize "${macports::prefix}/foo"]] ${macports::prefix}]
+    return [string map ${prefmap} [file normalize $filename]]
+}
+
 proc action_provides { action portlist opts } {
     # In this case, portname is going to be used for the filename... since
     # that is the first argument we expect... perhaps there is a better way
@@ -2207,13 +2213,19 @@ proc action_provides { action portlist opts } {
         return 1
     }
     foreach filename $portlist {
-        set file [file normalize $filename]
+        set file [macports::normalize $filename]
         if {[file exists $file] || ![catch {file type $file}]} {
             if {![file isdirectory $file] || [file type $file] eq "link"} {
+                set port0 [registry::file_registered $filename]
                 set port [registry::file_registered $file]
-                if { $port != 0 } {
+                if { $port0 != 0 } {
+                    if {![macports::ui_isset ports_quiet]} {
+                        puts -nonewline "$filename is provided by: "
+                    }
+                    puts $port0
+                } elseif { $port != 0 } {
                     if {![macports::ui_isset ports_quiet]} {
-                        puts -nonewline "$file is provided by: "
+                        puts -nonewline "$filename (${file}) is provided by: "
                     }
                     puts $port
                 } else {
